### PR TITLE
map context instances using `createContext` as key

### PR DESCRIPTION
**Pull request by phryneas**
@TobbeLino could you please try the CodeSandbox package build for this PR and give us feedback if it solves the problem?

```
npm i https://pkg.csb.dev/reduxjs/react-redux/commit/cd8a93bd/react-redux
```

### CODE REVIEW

**Comment by TobbeLino** [comment]
@phryneas Great! Yes, my test cases (and also our preact apps) run fine now with this PR!

File: src/components/Context.ts [hunk 1]
```diff
@@ -15,17 +15,20 @@ export interface ReactReduxContextValue<
   noopCheck: CheckFrequency
 }
 
-const ContextKey = Symbol.for(`react-redux-context-${ReactVersion}`)
-const gT = globalThis as { [ContextKey]?: Context<ReactReduxContextValue> }
+const ContextKey = Symbol.for(`react-redux-context`)
+const gT = globalThis as {
```
Comment (markerikson) [inline src/components/Context.ts:19]: Per https://github.com/reduxjs/react-redux/pull/2039#pullrequestreview-1524058384 , can we do some more checks to see if `globalThis` even exists?

Comment (phryneas) [reply src/components/Context.ts:19]: Given that Android 8.1 mentioned in that comment had it's official support from Google end on October 4, 2021, I would be very happy if we could avoid that - or at least only add it temporarily and remove it in a 9.0 release.

**Comment by phryneas** [comment]
Let's combine this with  #2050

**Pull request closed without merging**